From aa0e01436b3a27e5d368bbb12c05fc2f5195f189 Mon Sep 17 00:00:00 2001
From: Sriharsha Govinda Raj <sriharsha.govinda.raj@intel.com>
Date: Thu, 19 May 2016 15:42:46 +0530
Subject: [PATCH] EM/power_supply_charger:set MAX_THROTTLE_STATE for multiple
 chargers

MAX_THROTTLE_STATE was only set to the first charger which
was regsiter and was ignored for subsiquent chargers.
This patch supports setting MAX_THROTTLE_STATE for multiple
chargers.

Change-Id: I2a30ea670bbc64cbf4b141f6c05957a757dab718
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32401
Signed-off-by: Sriharsha <sriharsha.govinda.raj@intel.com>
Reviewed-on: https://android.intel.com:443/507394
---
 drivers/power/power_supply_charger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 26a9652..a4b548e 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -1019,10 +1019,10 @@ int power_supply_register_charger(struct power_supply *psy)
 
 	if (!psy_chrgr.is_cable_evt_reg && psy) {
 		ret = register_usb_notifier();
-		SET_MAX_THROTTLE_STATE(psy);
 		schedule_work(&notifier_work);
 	}
 	psy_chrgr.is_cable_evt_reg = true;
+	SET_MAX_THROTTLE_STATE(psy);
 	return ret;
 }
 EXPORT_SYMBOL(power_supply_register_charger);
-- 
1.9.1

